new_app.py

```diff
@@ -0,0 +1,247 @@
+                 'y_min',
+                 'y_max',
+                 'label'
+                ]]
+    new_df["left"] = new_df["x_min"]
+    new_df["top"] = new_df["y_min"]
+    new_df["width"] = abs(new_df["x_min"] - new_df["x_max"])
+    new_df["height"] = abs(new_df["y_min"] - new_df["y_max"])
+    new_df["type"] = "rect"
+    new_df["fill"] = "rgba(255, 0, 0, 0.2)"
+    new_df["stroke"] = "red"
+    new_df["result"] = False
+    new_df["user_reviewed"] = 0
+    return new_df
+
+def create_output_json(data_list):
+    return {
+        "objects": [
+            {
+                "type": item["type"],
+                "left": item["left"],
+                "top": item["top"],
+                "width": item["width"],
+                "height": item["height"],
+                "fill": item["fill"],
+                "stroke": item["stroke"],
+                "content": item["content"],
+                "label": item["label"],
+                "result": item["result"],
+                "file_name": item["filename"],
+                "user_reviewed": item["user_reviewed"]
+            }
+            for item in data_list
+        ]
+    }
+
+def resize_image(image, max_width, max_height):
+    aspect_ratio = float(image.size[1]) / float(image.size[0])
+    new_width = max_width
+    new_height = int(new_width * aspect_ratio)
+
+    if new_height > max_height:
+        new_height = max_height
+        new_width = int(new_height / aspect_ratio)
+
+    width_scale = float(new_width) / float(image.size[0])
+    height_scale = float(new_height) / float(image.size[1])
+
+    image.thumbnail((new_width, new_height))
+    return image, width_scale, height_scale
+
+def scale_bounding_box(saved_state, width_scale, height_scale):
+    is_single_object = len(saved_state["objects"]) == 1
+    for index, obj in enumerate(saved_state["objects"]):
+        saved_state["objects"][index]["left"] *= width_scale
+        saved_state["objects"][index]["top"] *= height_scale
+        saved_state["objects"][index]["width"] *= width_scale
+        saved_state["objects"][index]["height"] *= height_scale
+    return saved_state, is_single_object
+
+def next_page():
+    if st.session_state["file_index"] < len(files) - 1:
+        st.session_state["file_index"] += 1
+        if st.session_state["file_index"] in st.session_state["skipped_files"]:
+            next_page()
+        else:
+            st.experimental_rerun()
+    elif st.session_state["file_index"] == len(files) - 1:
+        st.session_state["file_index"] += 1
+        st.experimental_rerun()
+
+def previous_page():
+    if st.session_state["file_index"] > 0:
+        st.session_state["file_index"] -= 1
+        if st.session_state["file_index"] in st.session_state["skipped_files"]:
+            previous_page()
+        else:
+            st.experimental_rerun()
+    elif st.session_state["file_index"] == 0:
+        st.session_state["file_index"] -= 1
+        st.experimental_rerun()
+
+def main():
```

Comment:
it would be great to separate the layout from the rest by using functions for actions and putting all the layout in this function